src/test/java/ibm/jceplus/junit/base/BaseTestAESCopySafe.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright IBM Corp. 2023
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution.
+ */
+
+
+package ibm.jceplus.junit.base;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.Arrays;
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.junit.Assume;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+public class BaseTestAESCopySafe extends BaseTest {
+
+    private static final boolean DEBUG = false;
+    private static final int INPUT_LENGTH = 32; // should be a multiple of block size
+    private byte[] workingBuffer = new byte[INPUT_LENGTH + 512]; // Add plenty of room in our working buffer for overlap.
+    private SecretKey KEY = null;
+
+    private int[] OFFSETS = {0, 7, 8, 16, 32};
+
+    enum MODE { CBC, GCM };
+
+    protected int specifiedKeySize = 0;
+
+    public BaseTestAESCopySafe(String providerName) {
+        super(providerName);
+        this.specifiedKeySize = 128;
+    }
+
+    public BaseTestAESCopySafe(String providerName, int keySize) throws Exception {
+        super(providerName);
+        this.specifiedKeySize = keySize;
+        Assume.assumeTrue(javax.crypto.Cipher.getMaxAllowedKeyLength("AES") >= keySize);
+    }
+
+    public void testOverlappingBuffer() throws Exception {
+
+        KEY = new SecretKeySpec(new byte[specifiedKeySize / 8], "AES");
+
+        AlgorithmParameterSpec params = null;
+        for (MODE mode : MODE.values()) {
+            String transformation = "AES/" + mode.toString() + "/NoPadding";
+            Cipher c = Cipher.getInstance(transformation, providerName);
+            System.out.println("Testing " + transformation + " from provider: " + providerName);
+            for (int inputOffset : OFFSETS) {
+                for (int outputOffset : OFFSETS) {
+                    System.out.println("Mode: " + mode + " inputOffset: " + inputOffset + " outputOffset: " + outputOffset);
+                    System.out.println("Testing doFinal");
+                    doTest(c, params, inputOffset, outputOffset, mode, false);
+                    if (mode == MODE.CBC) {
+                        System.out.println("Testing update");
+                        doTest(c, params, inputOffset, outputOffset, mode, true);
+                    }
+                }
+            }
+        }
+    }
+
+    private void doTest(Cipher c, AlgorithmParameterSpec params, int inputOffset, int outputOffset, MODE mode, boolean isUpdate)
```

Comment:
The `params` parameter doesn't seem to be used any more. I think it should;d be removed.

Reply:
Yes agreed i did move `params` into the helper method `initCipher` here so it is no longer needed to be passed around. Adjusted this in new commit https://github.com/IBM/OpenJCEPlus/pull/295/commits/46a71f1f5e9d53cc94e3336b27fbf231deaeeab3 .